packages/block-editor/src/components/block-list/use-zoom-out-block-editing-mode.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { useBlockEditContext } from '../block-edit/context';
+
+export default function useZoomOutBlockEditingMode() {
+	const { clientId: rootClientId = '' } = useBlockEditContext();
+
+	const { unsetBlockEditingMode, setBlockEditingMode } =
+		useDispatch( blockEditorStore );
+
+	const { isZoomOutMode, sectionsContainerClientId, sectionsClientIds } =
+		useSelect( ( select ) => {
+			const {
+				__unstableGetEditorMode,
+				getSectionsContainerClientId,
+				getClientIdsOfDescendants,
+			} = select( blockEditorStore );
+
+			// TODO: We need a better API as using the post type
+			// in block editor package is not allowed.
+			const postType = select( 'core/editor' ).getCurrentPostType();
```

Comment:
I think this is a sign that logic doesn't belong in the `block-editor` package. Maybe we should try moving it to the `EditorProvider`, which already has similar handlers.

https://github.com/WordPress/gutenberg/blob/4cfaab6302c862ce264247ebef31b45729f04343/packages/editor/src/components/provider/index.js#L262-L267

Reply:
This is an oversight. We should not need it anymore.

Reply:
The selector might be an oversight, but I think it still makes sense to move the handler to a higher-level package.

Reply:
Which handler? The entire hook?

Reply:
This has been removed in the meantime.